proxyclient/hv.py

```diff
@@ -151,7 +165,13 @@ def handle_mmiotrace(self, data):
         else:
             t = "R"
 
-        print(f"[0x{evt.pc:016x}] MMIO: {t} 0x{evt.addr:x} = 0x{evt.data:x}")
+        try:
+            dev, start, size = self.mmio_lockup[evt.addr]
```

Comment:
And then, if you do the above, I think I'd rather just pre-populate mmio_lookup with all the ADT regs from the get go, instead of doing it lazily; this avoids having to do a whole ADT walk for every new range, we can just do it all in one walk.

Reply:
done